src/ws/login.html

```diff
@@ -78,6 +78,26 @@
               </div>
             </div>
 
+            <div id="language-group" class="form-group">
+              <label title="Language" for="server-field" class="col-sm-2 col-md-2 control-label" translate>Language</label>
+              <div class="col-sm-10 col-md-10 language-box">
+                <select id="language" class="form-control">
+                    <option value="en-us">English</option>
+                    <option value="ca-es">Catalan</option>
+                    <option value="da-dk">Dansk</option>
+                    <option value="de-de">Deutsch</option>
+                    <option value="es-es">Español</option>
+                    <option value="fr-fr">Français</option>
+                    <option value="ja-ja">日本語</option>
+                    <option value="pl-pl">Polski</option>
+                    <option value="pt-br">Portugueses</option>
+                    <option value="tr-tr">Türkçe</option>
+                    <option value="uk-ua">Українська</option>
+                    <option value="zh-cn">中文</option>
```

Comment:
This is copied from the static `"locales"` list in pkg/shell/manifest.json.in, right? So this would be okay for now, as it's not a regression.

But eventually login.js should fill this in dynamically from `locale -a` (I don't know a C interface for that, but I can look). E. g. cz and pt-PT are missing (from our po/LINGUAS), pt-br is Brazilian Portugese, there might be other available locales on the server, and some locales might not actually exist on the server.

Reply:
Filed as issue #8439 for generating the list at build time. Locales not being available on the server are still a problem, but not a big one (some called programs will croak warnings about unknown locales, but shouldn't be a big issue), so let's ignore that for now.